src/main/scala/org/broadinstitute/dsde/firecloud/service/MethodsService.scala

```diff
@@ -15,17 +23,58 @@ object MethodsService {
   val remoteMethodsUrl = FireCloudConfig.Agora.baseUrl + remoteMethodsPath
   val remoteConfigurationsPath = FireCloudConfig.Agora.authPrefix + "/configurations"
   val remoteConfigurationsUrl = FireCloudConfig.Agora.baseUrl + remoteConfigurationsPath
+  val remotePermissionsTemplate = FireCloudConfig.Agora.baseUrl + FireCloudConfig.Agora.authPrefix + "/%s/%s/%s/%s/permissions"
 }
 
-trait MethodsService extends HttpService with FireCloudDirectives {
+trait MethodsService extends HttpService with PerRequestCreator with FireCloudDirectives {
 
   lazy val log = LoggerFactory.getLogger(getClass)
 
   val localMethodsPath = "methods"
   val localConfigsPath = "configurations"
 
-  val routes: Route =
+  // Agora permissions that require special handling
+  val methodsAndConfigsACLOverrideRoute: Route =
+    path( "configurations|methods".r / Segment / Segment / IntNumber / "permissions") { ( agora_ent, namespace, name, snapshotId) =>
+      val url = getUrlFromBasePath(agora_ent, namespace, name, snapshotId)
+      get { requestContext =>
+        // pass to AgoraPermissionHandler
+        perRequest(requestContext,
+          Props(new AgoraPermissionActor(requestContext)),
+          AgoraPermissionHandler.Get(url))
+      } ~
+      post {
+        // explicitly pull in the json-extraction error handler from ModelJsonProtocol
+        handleRejections(entityExtractionRejectionHandler) {
+          // take the body of the HTTP POST and construct a FireCloudPermission from it
+          entity(as[List[FireCloudPermission]]) {
+            fireCloudPermissions => requestContext =>
+              perRequest(
+                requestContext,
+                Props(new AgoraPermissionActor(requestContext)),
+                AgoraPermissionHandler.Post(url, fireCloudPermissions.map(_.toAgoraPermission)))
+          }
+        }
+      } ~
+      delete {
+        complete(StatusCodes.MethodNotAllowed)
+      } ~
+      put {
+        complete(StatusCodes.MethodNotAllowed)
```

Comment:
Do we need to explicitly handle the put and delete cases? Won't spray do that for us if it can't find the path + method combination?